### PR TITLE
rewrite gossip callback

### PR DESCRIPTION
**Pull request by woutslakhorst**
The gossip callback is the first to remove the observer and use the universal notifier.
The others can't be done yet due to stoabs not being used yet in the DAG.
The new Notifier option on the state will replace all subscribers on the state eventually.

### CODE REVIEW

File: network/dag/state.go [hunk 1]
```diff
@@ -279,6 +326,13 @@ func (s *state) Start() error {
 		return err
 	}
 
+	// resume all notifiers
+	for _, subscriber := range s.notifiers {
```
Comment (gerardsn) [inline network/dag/state.go:330]: subscriber -> notifier

File: network/dag/interface.go [hunk 2]
```diff
@@ -54,6 +54,14 @@ type State interface {
...
 	// IsPresent returns true if a transaction is present in the DAG
 	IsPresent(context.Context, hash.SHA256Hash) (bool, error)
+	// Notifier creates a new Notifier.
+	// It lets another part of the application receive events on new transactions. When a new transaction is received
+	// the `receiver` function is called. Notifiers can be persistent and will survive restarts.
+	// The name is used to keep different notifiers apart.
+	// Filters can be used to receive specific transactions. Filters are added via the WithSelectionFilter() option.
+	// A Notifier should only be created during `configuration` step since the `start` step will redeliver all events that have not been delivered yet.
+	// Returns an error when the Notifier already exists
+	Notifier(name string, receiver ReceiverFn, filters ...NotifierOption) (Notifier, error)
```
Comment (gerardsn) [inline network/dag/interface.go:64]: Probably doesn't need to return the `Notifier`?

**Review by gerardsn — CHANGES_REQUESTED** [review]

File: network/dag/interface.go (hunk 2 above)
Comment (woutslakhorst) [reply network/dag/interface.go:64]: Not related to earlier comment, but I'll take it:
see nuts-foundation/go-stoabs#10, when the DB is accessible we can compare the TX with the DB

Comment (gerardsn) [reply network/dag/interface.go:64]: Sounds like a good solution

Comment (woutslakhorst) [reply network/dag/interface.go:64]: @gerardsn implemented

**Review by gerardsn — APPROVED** [review]

**Pull request merged**